pkg/es/config/config.go

```diff
@@ -70,34 +81,56 @@ func (p IndexPrefix) Apply(indexName string) string {
 // Configuration describes the configuration properties needed to connect to an ElasticSearch cluster
 type Configuration struct {
-	Servers                  []string               `mapstructure:"server_urls" valid:"required,url"`
-	RemoteReadClusters       []string               `mapstructure:"remote_read_clusters"`
-	Username                 string                 `mapstructure:"username"`
-	Password                 string                 `mapstructure:"password" json:"-"`
-	TokenFilePath            string                 `mapstructure:"token_file"`
-	PasswordFilePath         string                 `mapstructure:"password_file"`
-	AllowTokenFromContext    bool                   `mapstructure:"-"`
-	Sniffer                  bool                   `mapstructure:"sniffer"` // https://github.com/olivere/elastic/wiki/Sniffing
-	SnifferTLSEnabled        bool                   `mapstructure:"sniffer_tls_enabled"`
-	MaxDocCount              int                    `mapstructure:"-"` // Defines maximum number of results to fetch from storage per query
-	MaxSpanAge               time.Duration          `mapstructure:"-"` // configures the maximum lookback on span reads
-	Timeout                  time.Duration          `mapstructure:"-"`
-	BulkSize                 int                    `mapstructure:"-"`
-	BulkWorkers              int                    `mapstructure:"-"`
-	BulkActions              int                    `mapstructure:"-"`
-	BulkFlushInterval        time.Duration          `mapstructure:"-"`
-	Indices                  Indices                `mapstructure:"indices"`
-	ServiceCacheTTL          time.Duration          `mapstructure:"service_cache_ttl"`
-	AdaptiveSamplingLookback time.Duration          `mapstructure:"-"`
-	Tags                     TagsAsFields           `mapstructure:"tags_as_fields"`
-	Enabled                  bool                   `mapstructure:"-"`
-	TLS                      configtls.ClientConfig `mapstructure:"tls"`
-	UseReadWriteAliases      bool                   `mapstructure:"use_aliases"`
-	CreateIndexTemplates     bool                   `mapstructure:"create_mappings"`
-	UseILM                   bool                   `mapstructure:"use_ilm"`
-	Version                  uint                   `mapstructure:"version"`
-	LogLevel                 string                 `mapstructure:"log_level"`
-	SendGetBodyAs            string                 `mapstructure:"send_get_body_as"`
+	// Servers is a list of Elasticsearch servers. The strings must must contain full URLs
+	// (i.e. http://localhost:9200).
+	Servers []string `mapstructure:"server_urls" valid:"required,url"`
+	// RemoteReadClusters is a list of Elasticsearch remote cluster names for cross-cluster
+	// querying.
+	RemoteReadClusters []string       `mapstructure:"remote_read_clusters"`
+	Authentication     Authentication `mapstructure:"auth"`
+	// TokenFilePath contains the path to a file containing a bearer token.
+	// This flag also loads CA if it is specified.
```

Comment:
can you elaborate / code link?

Reply:
This actually doesn't look to be relevant anymore so I removed it. The CA File is loaded as part of the TLS configuration.